doc/python/performance.md

```diff
@@ -0,0 +1,298 @@
+
+Here is an example that creates a 100,000 point scatter plot using Plotly Express with WebGL rendering explicitly enabled.
+
+```python
+import plotly.express as px
+
+import pandas as pd
+import numpy as np
+
+np.random.seed(1)
+
+N = 100000
+
+df = pd.DataFrame(dict(x=np.random.randn(N),
+                       y=np.random.randn(N)))
+
+fig = px.scatter(df, x="x", y="y", render_mode='webgl')
+
+fig.update_traces(marker_line=dict(width=1, color='DarkSlateGray'))
+
+fig.show()
+```
+
+
+#### WebGL with 1,000,000 points with Graph Objects
+
+If Plotly Express does not provide a good starting point, it is also possible to use [the more generic `go.Scattergl` class from `plotly.graph_objects`](/python/graph-objects/).
+
+```python
+import plotly.graph_objects as go
+
+import numpy as np
+
+N = 1_000_000
+
+fig = go.Figure()
+
+fig.add_trace(
+    go.Scattergl(
+        x = np.random.randn(N),
+        y = np.random.randn(N),
+        mode = 'markers',
+        marker = dict(
+            line = dict(
+                width = 1,
+                color = 'DarkSlateGrey')
+        )
+    )
+)
+
+fig.show()
+```
+
+See https://plotly.com/python/reference/scattergl/ for more information and chart attribute options!
+
+## NumPy and NumPy Convertible Arrays for Improved Performance
+
+*New in Plotly.py version 6*
+
+Improve the performance of generating Plotly figures that use a large number of data points by using NumPy arrays and other objects that Plotly can convert to NumPy arrays, such as Pandas and Polars Series.
+
+Plotly.py uses Plotly.js for rendering, which supports typed arrays. In Plotly.py, NumPy array and NumPy-convertible arrays are base64 encoded before being passed to Plotly.js for rendering.
+
+### Arrays and Data Types Supported
+
+The following types of objects in Python are supported for improved performance:
+
+- Numpy `numpy.ndarray` objects.
+- Pandas Index, Pandas Series, Polars Series, and PyArrow Chunked Array objects. 
+- Array objects that can be converted to `numpy.ndarray` objects. i.e., they implement `"__array__"` or `"__array_interface__"` and return a `numpy.ndarray`.
+
+The following [array data types](https://numpy.org/devdocs/reference/arrays.scalars.html) are supported:
+
+- float32
+- float64
+- int8
+- uint8
+- int16
+- uint16
+- int32
+- uint32
+
+*If the array dtype is **int64** and **uint64**, often the default dtype for arrays in NumPy when no dtype is specified, those dtypes will be changed to other types internally by Plotly.py where possible. When working with NumPY directly, you can [specify the `dtype`](https://numpy.org/doc/stable/user/basics.types.html#array-types-and-conversions-between-types) when creating `ndarray` objects.
+
+### Unsupported Attributes
+
+Arrays passed to attributes with the following names are not supported:
```

Comment:
does "not supported" mean "they won't work"? or just "they won't get performance benefits"?

Reply:
just not supported for the performance benefits. updating